### PR TITLE
[Improvement-4069][server] When the tenant does not exist, the task execution should throw an exception

### PR DESCRIPTION
**Pull request by zhuangchong**
#4069

## What is the purpose of the pull request

When the workflow task node executes, if the tenant does not exist, the worker Task should not create the tenant, but throw an exception and the task ends.

## Brief change log

When the workflow task node executes, if the tenant does not exist, the Worker Task prints the log and returns it

## Verify this pull request
This pull request is already covered by existing tests

---
## pull request的目的是什么

工作流任务节点在执行时，如果租户不存在，worker task不应该去创建租户，而是抛出异常，任务结束。

## 简短的更改日志

工作流任务节点在执行时，如果租户不存在，worker task打印日志，返回

## 验证此pull request
此拉取请求已被现有测试覆盖

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/processor/TaskExecuteProcessor.java [hunk 1]
```diff
@@ -100,6 +98,12 @@ public void process(Channel channel, Command command) {
             logger.error("task execution context is null");
             return;
         }
+        // check if the OS user exists
+        if (!OSUtils.getUserList().contains(taskExecutionContext.getTenantCode())) {
+            logger.error("tenantCode:{} does not exist",taskExecutionContext.getTenantCode());
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/processor/TaskExecuteProcessor.java:103]: we can set the task status failed when the tenant code does not exist.

**Review by CalvinKirs — APPROVED** [review]
+1

**Review by lenboo — APPROVED** [review]
+1

**Comment by zhuangchong** [comment]
local test results:

image

**Pull request merged**